.clang-tidy

```diff
@@ -0,0 +1,27 @@
+---
+Checks:    
+      'modernize-*,
+       -modernize-use-equals-default,
+       -modernize-concat-nested-namespaces'
```

Comment:
So only using modernize?  What about cpp-core-guidelines?

Reply:
This is just a start. We will be enabled a bunch more including `cpp-core-guidelines` in follow up PRs

Reply:
Won't it be super disruptive to do this in steps?

Reply:
I don't think so. It is more manageable to break it up into small pieces imo.

Reply:
Keep in mind, this is very dissimilar to `clang-format`. format is set and forget it. tidy requires a lot more overheard for me to manually fix all the checks.